miden-tx/tests/test_miden_note_scripts.rs

```diff
@@ -1,166 +1,62 @@
-    notes::{Note, NoteOrigin, NoteScript},
-    Felt, StarkField, Word, ONE,
+    utils::collections::Vec,
+    Felt, StarkField,
 };
-use miden_stdlib::StdLibrary;
 use mock::{
     constants::{
-        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN, ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_1,
-        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_2, ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN,
-        ACCOUNT_ID_SENDER, DEFAULT_ACCOUNT_CODE,
-    },
-    mock::{
-        account::{mock_account_storage, MockAccountType},
-        notes::AssetPreservationStatus,
-        transaction::mock_inputs_with_existing,
+        ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN, ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN_2,
+        ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_ON_CHAIN, ACCOUNT_ID_SENDER,
+        DEFAULT_ACCOUNT_CODE,
     },
+    mock::account::mock_account_storage,
 };
 
-use miden_tx::{DataStore, TransactionExecutor};
-
-#[derive(Clone)]
-pub struct MockDataStore {
-    pub account: Account,
-    pub block_header: BlockHeader,
-    pub block_chain: ChainMmr,
-    pub notes: Vec<Note>,
-}
-
-impl MockDataStore {
-    pub fn with_existing(account: Option<Account>, consumed_notes: Option<Vec<Note>>) -> Self {
-        let (account, block_header, block_chain, consumed_notes) = mock_inputs_with_existing(
-            MockAccountType::StandardExisting,
-            AssetPreservationStatus::Preserved,
-            account,
-            consumed_notes,
-        );
-        Self {
-            account,
-            block_header,
-            block_chain,
-            notes: consumed_notes,
-        }
-    }
-}
+use miden_tx::TransactionExecutor;
 
-impl DataStore for MockDataStore {
-    fn get_transaction_data(
-        &self,
-        account_id: AccountId,
-        block_num: u32,
-        notes: &[NoteOrigin],
-    ) -> Result<(Account, BlockHeader, ChainMmr, Vec<Note>), miden_tx::DataStoreError> {
-        assert_eq!(account_id, self.account.id());
-        assert_eq!(block_num as u64, self.block_header.block_num().as_int());
-        assert_eq!(notes.len(), self.notes.len());
-        let origins = self
-            .notes
-            .iter()
-            .map(|note| note.proof().as_ref().unwrap().origin())
-            .collect::<Vec<_>>();
-        notes.iter().all(|note| origins.contains(&note));
-        Ok((
-            self.account.clone(),
-            self.block_header.clone(),
-            self.block_chain.clone(),
-            self.notes.clone(),
-        ))
-    }
-
-    fn get_account_code(
-        &self,
-        account_id: AccountId,
-    ) -> Result<ModuleAst, miden_tx::DataStoreError> {
-        assert_eq!(account_id, self.account.id());
-        Ok(self.account.code().module().clone())
-    }
-}
+mod common;
+use common::MockDataStore;
 
-// We test the Pay to ID script. So we create a note that can
-// only be consumed by the target account.
+/// P2ID TESTS
+/// ===============================================================================================
```

Comment:
minor nit: we use `///` only for doc comments. For section separators we should still use `//`.